Nodejs/Product/TypeScriptSourceMapReader/SourceMapReader.cs

```diff
@@ -0,0 +1,290 @@
+                // Read the source Map from sourceMapUri
+                encodedSourceMap = this.ReadEncodedSourceMap(sourceMapUri);
+            } else {
+                // Read source map using sourceMapText
+                StringReader stringReader = null;
+                try {
+                    stringReader = new StringReader(sourceMapText);
+                    encodedSourceMap = this.ReadEncodedSourceMap(stringReader);
+                } catch (Exception e) {
+                    throw new SourceMapReadFailedException(sourceMapUri.IsFile ? sourceMapUri.LocalPath : sourceMapUri.AbsoluteUri, e);
+                } finally {
+                    if (stringReader != null) {
+                        stringReader.Close();
+                    }
+                }
+            }
+
+            // Decode the source map for use
+            return DecodeSourceMap(encodedSourceMap, scriptFilePathOrUrl, sourceMapUri);
+        }
+        #endregion
+
+        #region Read encoded source map
+        /// <summary>
+        /// Gets the encoded source map (json) for the given script document
+        /// </summary>
+        /// <param name="scriptFilePathOrUrl">FilePath or Url of the js script file</param>
+        /// <param name="sourceMapUrl">url of the source map</param>
+        /// <param name="sourceMapUri">Returns Uri used to read the source map from</param>
+        /// <returns>EncodedSourceMap if the mapping present otherwise null, throws SourceMapReaderException if read fails</returns>
+        private EncodedSourceMap ReadEncodedSourceMap(string scriptFilePathOrUrl, string sourceMapUrl, out Uri sourceMapUri) {
+            EncodedSourceMap encodedSourceMap;
+            sourceMapUri = this.ReadSourceMap(scriptFilePathOrUrl, sourceMapUrl, out encodedSourceMap, textReader => this.ReadEncodedSourceMap(textReader));
+            return encodedSourceMap;
+        }
+
+        /// <summary>
+        /// Gets the encoded source map (json) for the given sourceMapUri
+        /// </summary>
+        /// <param name="sourceMapUri">Uri used to read the source map from</param>
+        /// <returns>EncodedSourceMap if the mapping present otherwise null, throws SourceMapReaderException if read fails</returns>
+        private EncodedSourceMap ReadEncodedSourceMap(Uri sourceMapUri) {
+            return this.ReadMapFile(sourceMapUri, textReader => this.ReadEncodedSourceMap(textReader));
+        }
+
+        /// <summary>
+        /// Gets the encoded source map from the sourceMapText reader
+        /// Note that this method doesnt catch any exceptions to convert into SourceMapRead failed exception
+        /// </summary>
+        /// <param name="sourceMapTextReader"></param>
+        /// <returns></returns>
+        private EncodedSourceMap ReadEncodedSourceMap(TextReader sourceMapTextReader) {
+            JsonTextReader jsonReader = null;
+            try {
+                jsonReader = new JsonTextReader(sourceMapTextReader);
```

Comment:
Same here, use a `using` statement instead of manually implementing the `finally` logic.